mopro-ffi/src/lib.rs

```diff
@@ -126,12 +126,8 @@ pub struct GenerateProofResult {
 macro_rules! app {
     () => {
         // These are mandatory imports for the uniffi to pick them up and match with UDL
-        use circom_prover::{
-            prover::{CircomProof, ProofLib},
-            witness::WitnessFn,
-            ProofCalldata, G1, G2,
-        };
-        use mopro_ffi::{GenerateProofResult, MoproError};
+        use circom_prover::witness::WitnessFn;
```

Comment:
I am still concern about the import
but we can fix this by this issue 
if the circom feature is disabled
https://github.com/zkmopro/mopro/blob/7d42ac161b0d3ce9f90314e39edab6cba0003dd9/mopro-ffi/Cargo.toml#L19-L20
Apps will not find the `circom_prover` dependency

but it can be fixed by this issue
https://github.com/zkmopro/mopro/issues/306

Reply:
Good point, I didn't think about that. I submitted [d3234fb](https://github.com/zkmopro/mopro/pull/312/commits/d3234fb605ab3b1ed3c57e124d811ac2206c4096) to fix this. The downside of this solution is caller (`test-e2e` needs to define `circom` feature in toml file). The other solution is to implement an empty struct like `circom_app!` and `halo2_app!`.

Reply:
@vivianjeng , I just realized, our current implementation also has this issue. For example, if we disable `circom` feature, it wll cause an error that `set_circom_circuits` can't be found. So, I think it's a good idea to address this issue in #306 as you mentioned.